attestation-agent/kbs_protocol/src/token_provider/mod.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+pub mod mock;
+pub use mock::*;
+
+use anyhow::*;
+use async_trait::async_trait;
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use jwt_simple::{
+    claims::JWTClaims,
+    prelude::{Clock, UnixTimeStamp},
+};
+use serde_json::Value;
+
+use crate::keypair::TeeKeyPair;
+
+#[async_trait]
+pub trait TokenProvider: Send + Sync {
+    /// Get token provisioned by Kbs and the tee pub key
+    ///
+    /// The returned value is a (Token, Private key) pair.
+    async fn get_token(&self) -> Result<(Token, TeeKeyPair)>;
+}
+
+#[derive(Clone, Debug)]
```

Comment:
We should implement `ToString` or some thing like for this struct, so that we can get the original content of JWT.